File: backend/app/models/ai_conversation.py
from datetime import datetime, timezone
from sqlalchemy import Column, String, Boolean, DateTime, Text
from ..database import Base

class AiConversation(Base):
    __tablename__ = "ai_conversations"

    id = Column(String, primary_key=True)
    user_id = Column(String(255), nullable=False, index=True)
    tenant_id = Column(String(50), default="default", index=True)
    title = Column(String(255), default="")
    messages = Column(Text, default="[]")
    active = Column(Boolean, default=True)
    created_at = Column(DateTime, default=lambda: datetime.now(timezone.utc))
    updated_at = Column(DateTime, default=lambda: datetime.now(timezone.utc), onupdate=lambda: datetime.now(timezone.utc))

File: dotnet-backend/src/Javaline.Commercial.Domain/Entities/AiConversation.cs
using System.ComponentModel.DataAnnotations.Schema;
using Javaline.Commercial.Domain.Common;

namespace Javaline.Commercial.Domain.Entities;

[Table("ai_conversations")]
public class AiConversation : BaseEntity
{
    [Column("user_id")]
    public string UserId { get; set; } = string.Empty;

    [Column("tenant_id")]
    public string TenantId { get; set; } = "default";

    [Column("title")]
    public string? Title { get; set; }

    [Column("messages")]
    public string Messages { get; set; } = "[]";

    [Column("active")]
    public bool Active { get; set; } = true;

    [Column("updated_at")]
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
}
